src/Http/Http.Extensions/src/IEndpointParameterMetadataProvider.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.Metadata;
+
+/// <summary>
+/// Indicates that a type provides a static method that returns <see cref="Endpoint"/> metadata when declared as the
+/// parameter type of an <see cref="Endpoint"/> route handler delegate.
+/// </summary>
+public interface IEndpointParameterMetadataProvider
+{
+    /// <summary>
+    /// Populates metadata for the related <see cref="Endpoint"/>.
```

Comment:
```suggestion
    /// Populates metadata for the related <see cref="Endpoint"/> parameter.
```


Wonder if we should add a remark that the `EndpointParameterMetadataContext.EndpointMetadata` list should be added to?